src/worker.rs

```diff
@@ -8,15 +8,47 @@ use std::{thread, time};
 
 use camunda_client::models::VariableValueDto;
 
-type VariablesMap = HashMap<String, VariableValueDto>;
+pub type VariablesMap = HashMap<String, VariableValueDto>;
+pub type HandlerResult = Result<VariablesMap, Box<dyn Error>>;
 
 pub fn wait(interval_seconds: u64) {
     thread::sleep(time::Duration::from_secs(interval_seconds));
 }
 
+pub struct Handler {
+    pub topic: String,
+    pub worker_id: String,
+    pub config: crate::config::Config,
+    pub action: Box<dyn Fn(Task) -> HandlerResult + Send + 'static>,
```

Comment:
I am not familiar with the syntax used in the type here.

how is this different than `pub action: dyn Fn(Task) -> HandlerResult` ?

Reply:
`pub action: dyn Fn(Task) -> HandlerResult` would not compile, Rust would complain that it doesn't have a size known at compile-time.

By wrapping the closure in a `Box`, we're hiding it behind a pointer, which has a known size. Also, it allows us to assign any closure with a matching signature while not having to deal with lifetimes.

Lastly, the whole struct must be [`Send`](https://doc.rust-lang.org/stable/std/marker/trait.Send.html) in order to work properly with `tokio`, and wrapping the closure in a `Box` is the only way that I know of to ensure that. It's also the reason why there's a `+ Send` in there.

For a better explanation, I also recommend to read this answer:
https://stackoverflow.com/a/27832320

Reply:
I've played with the types a bit more and the `Box` is gone now. It's likely still required on the call site, but it's not enforced in the library anymore.